src/nom.rs

```diff
@@ -602,9 +602,10 @@ pub fn float(input: &[u8]) -> IResult<&[u8],f32> {
     recognize!(
       tuple!(
         opt!(alt!(tag!("+") | tag!("-"))),
-        alt!(
-          delimited!(digit, tag!("."), opt!(digit))
-          | delimited!(opt!(digit), tag!("."), digit)
+        alt_complete!(
```

Comment:
Other streaming parsers I've seen, such as HTTP parsers, handle cases like this by returning an "I need more data" indication to the caller (ideally with some idea of "how much more data"), which has to obtain more data and call back into the parser.